src/crate/client/test_result.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+
+from crate import client
+from crate.client.result import BulkResponse
+from crate.client.test_support import setUpCrateLayerBaseline, tearDownDropEntitiesBaseline
+from crate.testing.settings import crate_host
+
+
+class BulkOperationTest(unittest.TestCase):
+
+    def setUp(self):
+        setUpCrateLayerBaseline(self)
+
+    def tearDown(self):
+        tearDownDropEntitiesBaseline(self)
+
+    def test_executemany_with_bulk_response(self):
+
+        connection = client.connect(crate_host)
+        cursor = connection.cursor()
+
+        # Run SQL DDL and a first valid INSERT operation.
+        cursor.execute("CREATE TABLE foobar (id INTEGER);")
+        _ = cursor.executemany("INSERT INTO foobar (id) VALUES (?)", [(1, ), (2, ), (3, ),])
+        cursor.execute("REFRESH TABLE foobar;")
+
+        # Run an invalid INSERT operation that will fail.
+        invalid_records = [(4, ), ("Hotzenplotz", )]
+        result = cursor.executemany("INSERT INTO foobar (id) VALUES (?)", invalid_records)
+
+        # Verify CrateDB response.
+        self.assertEqual(result, [{"rowcount": -2}, {"rowcount": -2}])
+
+        # Verify decoded response.
+        bulk_response = BulkResponse(invalid_records, result)
+        self.assertEqual(bulk_response.failed_records, invalid_records)
+
+        cursor.execute("REFRESH TABLE foobar;")
+        cursor.execute("SELECT * FROM foobar;")
+        result = cursor.fetchall()
+        self.assertEqual(sorted(result), [[1], [2], [3]])
```

Comment:
Currently, CrateDB seems to fail the second batch `[(4, ), ("Hotzenplotz", )]` completely. Is that normal?

Reply:
If there is syntax error, or error in analysis (e.g.: mismatching values with data types) the whole bulk fails before starting executing any of the inserts. See the issue and my comment: https://github.com/crate/crate/issues/16683#issuecomment-2376499787 for details.

Reply:
Thanks!